publish2cloud.py

```diff
@@ -307,11 +307,13 @@ def publish_to_remote_settings(config, section, chunknum, version):
         next_version = version.release[0] + 1
         if version.release[0] == shared_state.oldest_supported_version:
             # For all unsupported fx versions, we serve the oldest supported version
-            record_data['filter_expression'] = f'env.version|versionCompare("{version}") <= 0'
+            # Note: we have to compare against the nightly version, since 128.0a1 < 128.0b1 < 128.0
+            record_data['filter_expression'] = f'env.version|versionCompare("{version}a1") <= 0'
         else:
             # This filter_expression makes sure that a supported version is only given it's exact
             # versioned list
-            record_data['filter_expression'] = f'env.version|versionCompare("{version}") >= 0 && env.version|versionCompare("{next_version}") < 0'
+            # Note: we need to add a .0 to the next_version, since version.release[0] is an integer
+            record_data['filter_expression'] = f'env.version|versionCompare("{version}a1") >= 0 && env.version|versionCompare("{next_version}.0a1") < 0'
```

Comment:
Take 128 as an example.

IIUC, this won't work previously because the 128.0b1 is considered lower than 128. So, the beta version got the 127 lists, right? To fix this, we need to use the Nightly version name for 128 as the baseline, so the 128.0b1 can properly match the filter. correct?

Reply:
If this is the case, maybe we want to explain this explicitly in the comment, maybe with an example. So, people can understand it better.

Reply:
Updated the comment to have an example for our reasoning